src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -333,12 +333,10 @@ func (r *StatefulSetResource) obj() (k8sclient.Object, error) {
 							Args: []string{
 								"redpanda",
 								"start",
-								"--check=false",
```

Comment:
Why remove the `--check=false`? In a controlled environment like the cloud I don't think we need to run the checks each time.

Reply:
I was under impression that checks will failed to schedule pods if the environment is lacking some information. In the CI I found that this is not a case:
```
2021-04-13T16:20:02.193670112Z stderr F Detecting the current cloud vendor and VM
2021-04-13T16:20:02.193706215Z stderr F Could not detect the current cloud vendor
2021-04-13T16:20:02.193711316Z stderr F System check 'NTP Synced' failed with non-fatal error 'couldn't check NTP with timedatectl or ntpstat'
2021-04-13T16:20:02.215880047Z stderr F System check 'Dir '/var/lib/redpanda/data' IRQs affinity set' failed with non-fatal error 'no such file or directory'
2021-04-13T16:20:02.21591135Z stderr F no such file or directory
2021-04-13T16:20:02.21591645Z stderr F System check 'Dir '/var/lib/redpanda/data' scheduler tuned' failed with non-fatal error 'open : no such file or directory'
2021-04-13T16:20:02.215923951Z stderr F System check 'Fstrim systemd service and timer active' failed with non-fatal error 'dial unix /var/run/dbus/system_bus_socket: connect: no such file or directory'
2021-04-13T16:20:02.215933551Z stderr F no such file or directory
2021-04-13T16:20:02.215938052Z stderr F System check 'Dir '/var/lib/redpanda/data' nomerges tuned' failed with non-fatal error 'open : no such file or directory'
2021-04-13T16:20:02.215941852Z stderr F System check 'Dir '/var/lib/redpanda/data' IRQs affinity static' failed with non-fatal error 'no such file or directory'
2021-04-13T16:20:02.215945852Z stderr F System check 'RFS Table entries' failed with non-fatal error 'open /proc/sys/net/core/rps_sock_flow_entries: no such file or directory'
2021-04-13T16:20:02.215950053Z stderr F System check 'Clock Source' failed. Required: tsc, Current hyperv_clocksource_tsc_page
2021-04-13T16:20:02.215954353Z stderr F System check 'Data directory filesystem type' failed. Required: xfs, Current ext
2021-04-13T16:20:02.215958554Z stderr F System check 'Data partition free space [GB]' failed. Required: >= 10, Current 5.23
2021-04-13T16:20:02.215962154Z stderr F System check 'Dir '/var/lib/redpanda/data' IRQs affinity set' failed. Required: true, Current 
2021-04-13T16:20:02.215967554Z stderr F System check 'Dir '/var/lib/redpanda/data' IRQs affinity static' failed. Required: true, Current 
2021-04-13T16:20:02.215971555Z stderr F System check 'Dir '/var/lib/redpanda/data' nomerges tuned' failed. Required: true, Current 
2021-04-13T16:20:02.215975155Z stderr F System check 'Dir '/var/lib/redpanda/data' scheduler tuned' failed. Required: true, Current 
2021-04-13T16:20:02.215978855Z stderr F System check 'Free memory per CPU [MB]' failed. Required: 2048 per CPU, Current 50
2021-04-13T16:20:02.215982656Z stderr F System check 'Fstrim systemd service and timer active' failed. Required: true, Current 
2021-04-13T16:20:02.215986356Z stderr F System check 'I/O config file present' failed. Required: true, Current false
2021-04-13T16:20:02.215990056Z stderr F System check 'Max AIO Events' failed. Required: >= 1048576, Current 65536
2021-04-13T16:20:02.215993456Z stderr F System check 'Max syn backlog size' failed. Required: >= 4096, Current 512
2021-04-13T16:20:02.215997157Z stderr F System check 'NTP Synced' failed. Required: true, Current 
2021-04-13T16:20:02.216000857Z stderr F System check 'RFS Table entries' failed. Required: >= 32768, Current 
2021-04-13T16:20:02.216005157Z stderr F System check 'Swappiness' failed. Required: 1, Current 60
2021-04-13T16:20:02.216008858Z stderr F System check - PASSED
```

Reply:
I will bring that flag back.